controller/napi.go

```diff
@@ -332,11 +351,53 @@ func (r *reconcileIngress) reconcileNetworkAPI(ctx context.Context, ing *network
 
 	if len(ing.Status.LoadBalancer.Ingress) != 1 || ing.Status.LoadBalancer.Ingress[0].IP != vipIPStr {
 		ing.Status.LoadBalancer.Ingress = []corev1.LoadBalancerIngress{{IP: vipIPStr}}
-		err = r.client.Status().Update(ctx, ing)
+		err := r.client.Status().Update(ctx, ing)
 		if err != nil {
 			return err
 		}
 	}
 
 	return r.client.Status().Update(ctx, ing)
 }
+
+func (r *reconcileIngress) reconcileNetworkAPITakeOver(ctx context.Context, takeOverVIPName string, ing *networkingv1.Ingress, httpPool, httpsPool *networkapi.Pool) error {
+	lg := log.FromContext(ctx)
+
+	netapiCli := r.networkAPIClient
+	if netapiCli == nil {
+		netapiCli = networkapi.Client(r.cfg.NetworkAPIURL, r.cfg.NetworkAPIUsername, r.cfg.NetworkAPIPassword)
+	}
+
+	vip, err := netapiCli.GetVIP(ctx, takeOverVIPName)
+	if err != nil {
+		return err
+	}
+
+	if vip.IPv4 == nil {
+		return errors.New("no ipv4 found")
+	}
+
+	vipIP, err := netapiCli.GetIPByID(ctx, vip.IPv4.ID)
+	if err != nil && !networkapi.IsNotFound(err) {
```

Comment:
The `networkapi.IsNotFound` call along with the `vip name not found` error message should move up to after the `GetVIP` call.